client/log_client.go

```diff
@@ -196,12 +196,20 @@ func (c *LogClient) getAndVerifyLatestRoot(ctx context.Context, trusted *types.L
 		if err != nil {
 			return nil, err
 		}
+		// If this request hit a more out-of-date server instance than the GetSLR request,
+		// there may be an empty proof and an earlier SLR.
+		var proofRoot types.LogRootV1
+		if err := proofRoot.UnmarshalBinary(resp.GetSignedLogRoot().GetLogRoot()); err != nil {
```

Comment:
Are there tests for this?

Reply:
Good call -- in adding tests I found a bug.  Added 3 more commits at the end, PTAL.